packages/context/src/binding-inspector.ts

```diff
@@ -215,10 +220,16 @@ export function createBindingFromClass<T = unknown>(
   cls: Constructor<T | Provider<T>>,
   options: BindingFromClassOptions = {},
 ): Binding<T> {
+  debug('create binding from class %s with options', cls.name, options);
   const templateFn = bindingTemplateFor(cls);
   let key = options.key;
   if (!key) {
-    key = buildBindingKey(cls, options);
+    try {
+      key = buildBindingKey(cls, options);
+    } catch (err) {
+      err.message += ` (while building binding key for class ${cls.name})`;
```

Comment:
Is mutating the error message too invasive?

Reply:
Cross-posting from https://github.com/strongloop/loopback-next/pull/3617#discussion_r319084318:

> > Is there a better way than mutating the error message?
> 
> I am not aware of any. It's important to preserve the original stack trace and AFAIK, that requires us to mutate the original error object.

Do you have any specific concern in mind? I am not aware of any problems that could be caused by editing error messages.

Reply:
I was looking for a more systematic solution like https://stackoverflow.com/questions/42754270/re-throwing-exception-in-nodejs-and-not-losing-stack-trace. But I'm fine to change the message for this PR.